libs/journeys/ui/src/components/Card/Card.tsx

```diff
@@ -187,17 +188,18 @@ export function Card({
    */
   const handleFormSubmit = async (values: FormikValues): Promise<void> => {
     if (variant !== 'default' && variant !== 'embed') return
-    const heading =
-      activeBlock != null
-        ? getStepHeading(activeBlock.id, activeBlock.children, treeBlocks, t)
-        : t('None')
 
     const submissionPromises = textResponseBlocks.map((block) => {
       const blockId = block.id
       const responseValue = values[blockId]
       if (!responseValue || responseValue?.trim() === '')
         return Promise.resolve(null)
 
+      const heading =
+      activeBlock != null
+        ? (getTextResponseLabel(activeBlock.children, blockId) ??
```

Comment:
```suggestion
        ? (getTextResponseLabel(block) ??
```
just pass in the block itself, we don't need the children